homeassistant/components/hassmpris/config_flow.py

```diff
@@ -0,0 +1,206 @@
+                step_id="zeroconf_confirm",
+                data_schema=vol.Schema(
+                    {
+                        vol.Required(
+                            "host", description={"suggested_value": self._host}
+                        ): str,
+                    }
+                ),
+                errors={},
+            )
+        self._host = user_input["host"]
+        # Do not proceed if the device is already configured.
+        await self.async_set_unique_id(self._host)
+        self._abort_if_unique_id_configured()
+        return await self.async_step_pairing()
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if not user_input:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=STEP_USER_DATA_SCHEMA,
+                errors={},
+            )
+
+        self._host = user_input["host"]
+        # Do not proceed if the device is already configured.
+        await self.async_set_unique_id(self._host)
+        self._abort_if_unique_id_configured()
+        return await self.async_step_pairing()
+
+    async def async_step_reauth(self, user_input: dict[str, Any] | None = None):
+        """Handle the reauth step."""
+        assert user_input, "not possible"
+        self._host = user_input["host"]
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(self, user_input: dict[str, Any] | None = None):
+        """Handle the confirmation of reauth."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="reauth_confirm",
+                data_schema=vol.Schema({}),
+            )
+        self._client_cert = None
+        return await self.async_step_pairing()
+
+    async def async_step_pairing(
+        self,
+        user_input: dict[str, Any] | None = None,
+    ):
+        """Handle the pairing step of auth."""
+        assert not self._client_cert
+        (
+            csr,
+            key,
+        ) = pskca.create_certificate_signing_request()
+        self._client_key = key.private_bytes(
+            encoding=serialization.Encoding.PEM,
+            format=serialization.PrivateFormat.TraditionalOpenSSL,
+            encryption_algorithm=serialization.NoEncryption(),
+        ).decode("ascii")
+        self._cakes_client = hassmpris_client.AsyncCAKESClient(
+            self._host,
+            40052,
+            csr,
+        )
+        try:
+            ecdh = await self._cakes_client.obtain_verifier()
+        except hassmpris_client.Timeout:
+            return self.async_abort(reason="timeout_connect")
+        except hassmpris_client.ClientException:
+            return self.async_abort(reason="cannot_connect")
+        emojis = emoji(ecdh.derived_key, 6)
+
+        return self.async_show_form(
+            step_id=STEP_CONFIRM,
```

Comment:
in most places in HA, the steps are not used as constants, but if you do, better to be consistent across all steps

Reply:
I see no ConfigFlow steps given as strings, only as constants.